uniqueids/admin.py

```diff
@@ -54,22 +54,59 @@ class PersonnelUploadAdmin(admin.ModelAdmin):
     list_display = ["import_type", "created_at", "valid", "error"]
     exclude = ["valid", "error"]
 
-    def validate_keys(self, record, import_type):
-        required_keys = [
-            "address_type", "address", "preferred_language",
-            "receiver_role", "uniqueid_field_name",
-            "uniqueid_field_length", "name", "surname"]
+    required_keys = [
+        "address_type", "address", "preferred_language",
+        "receiver_role", "uniqueid_field_name",
+        "uniqueid_field_length", "name", "surname"]
 
-        required_keys_type = {
-            PersonnelUpload.CORP_TYPE: ["community"],
-            PersonnelUpload.PERSONNEL_TYPE: ["role", "facility_name", "state"]
-        }
+    required_keys_type = {
+        PersonnelUpload.CORP_TYPE: ["community"],
+        PersonnelUpload.PERSONNEL_TYPE: ["role", "facility_name", "state"]
+    }
 
-        required = set(required_keys + required_keys_type[import_type])
+    def validate_keys(self, record, import_type):
+        required = set(
+            self.required_keys + self.required_keys_type[import_type])
         missing = required - set(record.keys())
 
         return missing
 
+    def validate_values(self, record, import_type):
+        missing = set()
+        required = set(
+            self.required_keys + self.required_keys_type[import_type])
+
+        for key, value in record.items():
+            if key in required and not value:
+                missing.add(key)
+            elif key == "address_type" and value != "msisdn":
+                missing.add(key)
+            elif (key == "preferred_language" and
+                    value not in settings.LANGUAGES):
+                missing.add(key)
+            elif (key == "uniqueid_field_name" and
+                    value not in ['personnel_code', 'corp_code']):
+                missing.add(key)
+            elif (key == "uniqueid_field_length" and not value.isdigit() and
+                    value.find('-') == -1):
+                missing.add(key)
+
+        return missing
+
+    def validate_address(self, record):
+        msisdn = utils.normalize_msisdn(record["address"], '234')
+
+        identities = utils.search_identities(
+            "details__addresses__{}".format(record["address_type"]), msisdn)
+
+        for key in identities:
+            return True
+
+        if len(msisdn) != 13:
+            return True
+
+        return False
```

Comment:
At first look this function is a bit confusing, because it's returning `True` if the address is invalid, and `False` if the address is valid. Maybe we can switch those two around, and `not` the result when we use it below?